src/Translumo.TTS/TtsFactory.cs

```diff
@@ -9,24 +9,33 @@ public class TtsFactory
     {
         private readonly LanguageService _languageService;
         private readonly PythonEngineWrapper _pythonEngine;
+        private readonly IObserverAvailableVoices _observerAvailableVoices;
         private readonly ILogger _logger;
 
-        public TtsFactory(LanguageService languageService, PythonEngineWrapper pythonEngine, ILogger<TtsFactory> logger)
+        public TtsFactory(LanguageService languageService, PythonEngineWrapper pythonEngine, IObserverAvailableVoices observerAvailableVoices, ILogger<TtsFactory> logger)
         {
             _languageService = languageService;
             _pythonEngine = pythonEngine;
+            _observerAvailableVoices = observerAvailableVoices;
             _logger = logger;
         }
 
-        public ITTSEngine CreateTtsEngine(TtsConfiguration ttsConfiguration) =>
-            ttsConfiguration.TtsSystem switch
+        public ITTSEngine CreateTtsEngine(TtsConfiguration ttsConfiguration)
+        {
+            ITTSEngine ttsEngine = ttsConfiguration.TtsSystem switch
             {
                 TTSEngines.None => new NoneTTSEngine(),
                 TTSEngines.WindowsTTS => new WindowsTTSEngine(GetLangCode(ttsConfiguration)),
                 TTSEngines.SileroTTS => new SileroTTSEngine(_pythonEngine, GetLangCode(ttsConfiguration)),
                 _ => throw new NotSupportedException()
             };
 
+            var voices = ttsEngine.GetVoices();
+            _observerAvailableVoices.UpdateVoice(voices);
```

Comment:
Factory shouldn't affect any external components (single reposnobility is only to create new component - TTSEngine), let's put this update in ProcessingService